module/button.py

```diff
@@ -1,18 +1,67 @@
 from telegram import Update
 from telegram.ext import CallbackContext
+from module.add_book import add_book
+from module.add_item import add_item
 from module.create_connection import create_connection
-from module.shared import DB_PATH
+from module.shared import DB_PATH, error_message
+from module.add_book import add_book
+from module.add_item import add_item
+from module.manage_requests import delete_request, update_similar_requests
 
 
 def button(update: Update, context: CallbackContext) -> None:
     chat_id = update.effective_chat.id
     query = update.callback_query
-    conn = create_connection(DB_PATH)
-    if conn:
+    
+    if query.data.split(';')[0] == 'delete':
+        conn = create_connection(DB_PATH)
+        if conn:
+            cur = conn.cursor()
+            q = query.data.split(';')[1]
+            cur.execute("DELETE FROM Market WHERE rowid=?", (q,))
+            query.edit_message_text(text=f"Eliminazione del libro selezionato...")
+            conn.commit()
+            context.bot.send_message(chat_id, "Libro eliminato.")
+        else:
+            query.edit_message_text(text=f"Si è verificato un problema nella lettura del database.")
+        conn.close()
+    
+    if query.data.split(';')[0] == 'new_request':
+        vote = query.data.split(';')[1]
+        row_id = query.data.split(';')[2]
+        
+        conn = create_connection(DB_PATH)
+        if not conn:
+            context.bot.send_message(chat_id, error_message)
+            return
         cur = conn.cursor()
-        cur.execute("DELETE FROM Market WHERE rowid=?", (query.data,))
-        query.edit_message_text(text=f"Eliminazione del libro selezionato...")
-        conn.commit()
-        context.bot.send_message(chat_id, "Libro eliminato.")
-    else:
-        query.edit_message_text(text=f"Si è verificato un problema nella lettura del database.")
+        cur.execute("SELECT rowid, * FROM Requests WHERE rowid=?", (row_id,))
+        rows = cur.fetchall()
+        
+        if rows:
+
+            chat_id = rows[0][1]
+
+            if vote == 'Y':
+                isbn = rows[0][2]
+                title = rows[0][3]
+                authors = rows[0][4]
+                username = rows[0][5]
+                price = rows[0][6]
```

Comment:
```suggestion
_, __, isbn, title, authors, username, price = rows[0]
```

Reply:
would be better to set chat_id here and make the `chat_id = rows[0][1]` only inside `if vote == 'N'`

Reply:
So you suggest to write `_, chat_id, isbn, title, authors, username, price = rows[0]` in the `vote = Y` and put the `chat_id = rows[0][1]` in the `vote = N`?
By the way, chat_id is needed in both the if clauses, i thought it was a good idea to put it there